File: Online-Judge/timus/solutions/1052.cpp
#include <bits/stdc++.h>

using namespace std;


int main() {
    int n;
    cin >> n;

    vector<pair<int, int>> dots;
    for (int i = 0; i < n; i++) {
        int x, y;
        cin >> x >> y;
        dots.push_back({x, y});
    }

    int ans = 0;

    for (auto i: dots) {
        for (auto j: dots) {
            if (i == j) continue;
            pair<int, int> ab = {i.first - j.first, i.second - j.second};
            int s = 0;
            for (auto k: dots) {
                pair<int, int> bc = {j.first - k.first, j.second - k.second};
                int vec = ab.second * bc.first - ab.first * bc.second;
                if (vec == 0)
                    s += 1;
            ans = max(ans, s);
            }
        }
    }
    cout << ans;
}

File: Online-Judge/timus/solutions/1052.py
n = int(input())

dots = [list(map(int, input().split())) for _ in range(n)]

ans = 0

for i in dots:
    for j in dots:
        if i == j:
            continue
        ab = (i[0] - j[0], i[1] - j[1])
        s = 0
        for k in dots:
            bc = (j[0] - k[0], j[1] - k[1])
            vec = ab[1] * bc[0] - ab[0] * bc[1]
            if abs(vec) < 1e-7:
                s += 1
        ans = max(ans, s)

print(ans)
